kafka/README.md

```diff
@@ -144,17 +144,65 @@ az role assignment create --role "Key Vault Crypto User" --assignee <alias>@micr
 
 NOTE: Only the subscription owner can setup role access for AKV/mHSM, so if you are seeing authorization related error messages during role access setup steps, please seek out the proper personel to setup role access. 
 
-#### Install Kafka Cluster 
+
+#### Create Kafka Cluster 
+
+**NOTE**: Due to security issues, we have deprecated running Kafka cluster on k8s using Strimzi. The preferred kafka cluster in this example is Microsoft provisioned Azure Event Hub. However, we still keep the old instruction on how to run this demo using Kafka cluster on k8s using Strimzi. See the following: 
+
+**Create Strimzi Kafka Cluster (Deprecated):** 
 
 Install the Kafka cluster in the Kafka namespace using the following command or following the instructions [here](https://strimzi.io/quickstarts/)
 
 ```bash 
-kubectl create namespace kafka
 kubectl create -f 'https://strimzi.io/install/latest?namespace=kafka' -n kafka
 # Apply the `Kafka` Cluster CR file
 kubectl apply -f https://strimzi.io/examples/latest/kafka/kafka-persistent-single.yaml -n kafka 
 ```
 
+**Create Azure Event Hub Resource:**
+
+```bash 
+export EVENT_HUB_NAMESPACE=kafka-demo-ehubns
+export EVENT_HUB_NAME=kafka-demo-topic
+az eventhubs namespace create --name $EVENT_HUB_NAMESPACE --resource-group $RESOURCE_GROUP --location $LOCATION --enable-kafka true --enable-auto-inflate false
+az eventhubs eventhub create --name $EVENT_HUB_NAME --resource-group $RESOURCE_GROUP --namespace-name $EVENT_HUB_NAMESPACE --partition-count 10
+```
+
+#### Generate Keys ####
+This step prepare env vars for running setup-key.sh to generate the keys used for encrypting kafka messgages. 
+These env vars are also necessary for generating consumer pod yaml file if you are using Azure Event Hub. 
+Thus, run these commands regardless using Azure Event Hub or Strimzi Kafka cluster. 
+
+```bash
+$ export MAA_ENDPOINT="sharedeus.eus.attest.azure.net"
+# <akv/mHSM url> should have the following format
+# <akv-name>.vault.azure.net or <mHSM-name>.managedhsm.azure.net
+$ export AZURE_AKV_RESOURCE_ENDPOINT= <akv-name>.vault.azure.net or <mHSM-name>.managedhsm.azure.net
+```
+
+#### Generate Consumer Pod YAML Files (Only If Using Azure Event Hub Resource)
+
+```bash 
+$ result=$(az eventhubs namespace authorization-rule list --resource-group $RESOURCE_GROUP --namespace-name $EVENT_HUB_NAMESPACE)
+$ export EVENT_HUB_AUTH_RULE_NAME=$(echo "$result" | jq -r '.[0].name')
+$ result=$(az eventhubs namespace authorization-rule keys list --resource-group $RESOURCE_GROUP --namespace-name $EVENT_HUB_NAMESPACE --name $EVENT_HUB_AUTH_RULE_NAME)
+$ export EVENTHUB_CONNECTION_STRING="$(echo "$result" | jq -r '.primaryConnectionString')"
+$ export EVENTHUB_BROKER=$EVENT_HUB_NAMESPACE.servicebus.windows.net:9093
+$ export SIDECAR_IMAGE=""
+$ export CONSUMER_IMAGE=""
```

Comment:
we should probably put the default images here or tell the customer above to fill in these with the names of the images they created in another step